server.go

```diff
@@ -602,13 +752,18 @@ func (s *Server) Stop() {
 	s.lis = nil
 	cs := s.conns
 	s.conns = nil
+	for c := range s.hconns {
+		go c.Close()
```

Comment:
Surprising who? This code is only used by tests, and the tests don't synchronize it with anything else.

Who else is using this? grpc-go really needs a graceful shutdown mechanism, and this isn't it (or at least not yet). I don't want to fix that in this bug, but Stop by itself is pretty useless today. It's mostly a cleanup mechanism for tests, since normal servers don't call Stop. They'd really want graceful shutdown if they know they're going to shut down (else they can just crash/exit).